File: fsharp/pangram/Iteration1/Pangram.fs
module Pangram

let alphabet: char Set = Set.ofSeq "abcdefghijklmnopqrstuvwxyz"

let isPangram (input: string): bool =
    input
    |> fun s -> s.ToLower()
    |> Set.ofSeq
    |> Set.isSubset alphabet

File: haskell/pangram/Iteration1/src/Pangram.hs
module Pangram (isPangram) where

import Data.Char
import Data.List

-- Iteration 1:
-- isPangram :: String -> Bool
-- isPangram text = sortedChars == checksum
--     where
--         allChars       = filter allowedChar text
--         distinctChars  = nub allChars
--         lowercaseChars = map toLower distinctChars
--         sortedChars    = sort lowercaseChars

-- Iteration 2
isPangram :: String -> Bool
isPangram text = sort (map toLower (nub (filter allowedChar text))) == checksum

checksum :: String
checksum = "abcdefghijklmnopqrstuvwxyz"

allowedChar :: Char -> Bool
allowedChar c = isAlpha c && isAscii c